test/integration/quota/quota_test.go

```diff
@@ -593,3 +597,119 @@ func newService(name string, svcType v1.ServiceType, allocateNodePort bool) *v1.
 		},
 	}
 }
+
+func TestQuotaLimitResourceClaim(t *testing.T) {
+	tCtx := ktesting.Init(t)
+
+	// Set up an API server
+	_, kubeConfig, tearDownFn := framework.StartTestServer(tCtx, t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
+			opts.Admission.GenericAdmission.DisablePlugins = []string{"ServiceAccount"}
+			opts.APIEnablement.RuntimeConfig = flag.ConfigurationMap{
+				resourceapi.SchemeGroupVersion.String(): "true",
+			}
+		},
+	})
+	defer tearDownFn()
+
+	clientset := clientset.NewForConfigOrDie(kubeConfig)
+
+	informers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+
+	discoveryFunc := clientset.Discovery().ServerPreferredNamespacedResources
+	listerFuncForResource := generic.ListerFuncForResourceFunc(informers.ForResource)
+	qc := quotainstall.NewQuotaConfigurationForControllers(listerFuncForResource)
+	informersStarted := make(chan struct{})
+	resourceQuotaControllerOptions := &resourcequotacontroller.ControllerOptions{
+		QuotaClient:               clientset.CoreV1(),
+		ResourceQuotaInformer:     informers.Core().V1().ResourceQuotas(),
+		ResyncPeriod:              controller.NoResyncPeriodFunc,
+		InformerFactory:           informers,
+		ReplenishmentResyncPeriod: controller.NoResyncPeriodFunc,
+		DiscoveryFunc:             discoveryFunc,
+		IgnoredResourcesFunc:      qc.IgnoredResources,
+		InformersStarted:          informersStarted,
+		Registry:                  generic.NewRegistry(qc.Evaluators()),
+	}
+	resourceQuotaController, err := resourcequotacontroller.NewController(tCtx, resourceQuotaControllerOptions)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	go resourceQuotaController.Run(tCtx, 2)
+
+	// Periodically the quota controller to detect new resource types
+	go resourceQuotaController.Sync(tCtx, discoveryFunc, 30*time.Second)
+
+	informers.Start(tCtx.Done())
+	close(informersStarted)
+
+	wg := new(sync.WaitGroup)
+	namespaces := 500
+	for range namespaces {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			ns := framework.CreateNamespaceOrDie(clientset, "quota-"+utilrand.String(5), t)
```

Comment:
@pohly My intent with this test was to mimic the flaky e2e test, but essentially running a bunch of those concurrently in different namespaces. Does this seem like a fair representation of that?